core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -403,15 +403,22 @@ private JournalEntry getContainerIdJournalEntry() {
    * @param tierAlias the alias of the storage tier where the worker is committing the block to
    * @param blockId the committing block id
    * @param length the length of the block
+   * @throws NoWorkerException if the workerId is not active
    */
   public void commitBlock(long workerId, long usedBytesOnTier, String tierAlias, long blockId,
-      long length) {
+      long length) throws NoWorkerException {
     LOG.debug("Commit block from workerId: {}, usedBytesOnTier: {}, blockId: {}, length: {}",
         workerId, usedBytesOnTier, blockId, length);
 
     long counter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
 
     MasterWorkerInfo worker = mWorkers.getFirstByField(mIdIndex, workerId);
+    // TODO(peis): Check lost workers as well.
```

Comment:
I think this is a strong signal telling the master that this worker is alive now. We do this in BlockMaster#getWorkerId